src/pulselistener/pulselistener/listener.py

```diff
@@ -132,20 +133,42 @@ def should_run_risk_analysis(self, build):
             await self.bus.send(QUEUE_CODE_REVIEW, build)
 
 
-class HookCodeCoverage(PulseHook):
+class HookCodeCoverage(object):
     '''
     Taskcluster hook handling the code coverage
     '''
     def __init__(self, configuration, bus):
         assert 'hookId' in configuration
         self.triggered_groups = set()
-        super().__init__(
-            configuration.get('hookGroupId', 'project-releng'),
-            configuration['hookId'],
-            'exchange/taskcluster-queue/v1/task-group-resolved',
-            '#',
-            bus,
-        )
+        self.group_id = configuration.get('hookGroupId', 'project-releng')
+        self.hook_id = configuration['hookId']
+        self.bus = bus
+
+        # Setup TC services
+        self.queue = taskcluster.get_service('queue')
+        self.hooks = taskcluster.get_service('hooks')
+
+    async def run(self):
+        '''
+        Main consumer, running queued payloads from the pulse listener
+        '''
+        while not self.bus.queues[QUEUE_MONITORING].full():
```

Comment:
Why only while the monitoring queue is not full?

Reply:
If it's just because of the test, you could probably bring it back to `while True`, then in the test you run `run` without waiting for it to finish and you just wait for the message on the monitoring queue